### ISSUE TITLE
Crash triggered by scrollToBottomAnimated

### ISSUE THREAD

**Issue by nash-pwnage** [jessesquires/JSQMessagesViewController#1150]
I am using JSQMessagesVC on the latest develop branch. It has proved to be a really great library for my development. In my app, I flex the scrolling logic to enable scroll only when the last received message cell is on screen, if not avoid scrolling. So whenever I receive a message and i perform the above check and call scrollToBottomAnimated, i get a crash with the following stacktrace in [JSQMessagesCollectionViewFlowLayout layoutAttributesForItemAtIndexPath:]. 

The issue originates only on an Iphone 5s device running 8.2. I have tested the same code with 8.4 on Iphone 5, 5s, 6. But i only see this crash on 8.2

Stacktrace:
````
{
    Name = NSRangeException;
    Reason = "*** -[__NSArrayM objectAtIndex:]: index 21 beyond bounds [0 .. 20]";
    StackTrace =     (
        "#0  CoreFoundation                      0X0000000185E2E530 <redacted>()",
        "#1  libobjc.A.dylib                     0X0000000196DB00E4 objc_exception_throw()",
        "#2  CoreFoundation                      0X0000000185D1377C -[__NSArrayM removeObjectAtIndex:]",
        "#3  UIKit                               0X000000018AD018D4 -[_UIFlowLayoutSection frameForItemAtIndexPath:]",
        "#4  UIKit                               0X000000018ACD830C -[UICollectionViewFlowLayout _frameForItem:inSection:usingData:]",
        "#5  UIKit                               0X000000018ACD66CC -[UICollectionViewFlowLayout layoutAttributesForItemAtIndexPath:usingData:]",
        "#6  JSQMessagesViewController           0X0000000101A13AC8 -[JSQMessagesCollectionViewFlowLayout layoutAttributesForItemAtIndexPath:]",
        "#7  JSQMessagesViewController           0X0000000101A150C0 -[JSQMessagesCollectionViewFlowLayout sizeForItemAtIndexPath:]",
        "#8  JSQMessagesViewController           0X0000000101A06BF8 -[JSQMessagesViewController scrollToBottomAnimated:]",)
}
````

**Comment by nash-pwnage** [comment]
We figured this out. Apparently iOS invalidate layout implementation details changed under the covers. My initially placed `[self.collectionView.collectionViewLayout invalidateLayout]` did not trigger an update in the `collectionView:numberOfItemsInSection:` in iOS 8.2, 

But interestingly enough it did in iOS 8.4.

I fixed both the versions by calling  `[self.collectionView.collectionViewLayout invalidateLayoutWithContext:[JSQMessagesCollectionViewFlowLayoutInvalidationContext context]];`
